File: CodeWars_Python/invert_values.py
#!/usr/bin/env python

# Invert a given list of integer values.
#
# Python/JS:
#
# invert([1,2,3,4,5]) == [-1,-2,-3,-4,-5]
# invert([1,-2,3,-4,5]) == [-1,2,-3,4,-5]
# invert([]) == []


def invert(lst):
    return [-l for l in lst]


print(invert([1, 2, 3, 4, 5]))  # [-1,-2,-3,-4,-5])
print(invert([1, -2, 3, -4, 5]))  # [-1,2,-3,4,-5])
print(invert([]))  # [])

File: CodeWars_Ruby/invert_values.rb
#!/usr/bin/env ruby

# Invert a given list of integer values.
#
# Python/JS/Ruby:
#
# invert([1,2,3,4,5]) == [-1,-2,-3,-4,-5]
# invert([1,-2,3,-4,5]) == [-1,2,-3,4,-5]
# invert([]) == []

def invert(lst)
  lst.map { |l| l * -1 }
end

p invert([1, 2, 3, 4, 5]) # [-1,-2,-3,-4,-5])
p invert([1, -2, 3, -4, 5]) # [-1,2,-3,4,-5])
p invert([]) # [])
